src/main/java/lotto/model/BunchOfLotto.java

```diff
@@ -0,0 +1,25 @@
+package lotto.model;
+
+import lotto.util.BunchOfLottoGenerator;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class BunchOfLotto {
+    private final List<Lotto> bunchOfLotto;
+
+    public BunchOfLotto(int purchasedLottoCount) {
+        this.bunchOfLotto = BunchOfLottoGenerator.makeBunchOfLotto(purchasedLottoCount);
+    }
+
+    public List<Lotto> getBunchOfLotto() {
+        return Collections.unmodifiableList(bunchOfLotto);
+    }
+
+    public Prizes makeRewards(WinningLotto winningLotto) {
+        return new Prizes(bunchOfLotto.stream()
+                .map((lotto) -> Reward.getReward(winningLotto.makeWinningState(lotto)))
+                .collect(Collectors.toList()));
```

Comment:
한 번에 너무 많은 동작이 일어나는 것 같아요. 아래와 같이 로직을 분리해보면 어떨까요?  😃

 ```suggestion
        List<Reward> prizes = bunchOfLotto.stream()
            .map(lotto -> winningLotto.makeWinningState(lotto))
            .map(winningState -> Reward.getReward(winningState))
            .collect(Collectors.toList());
        return new Prizes(prizes);
```

개인적인 생각이지만, `WinningState`로 변환하는 과정이 불필요해 보여요. 
WinningLotto와 Lotto를 비교함으로써, 바로 `Reward`를 추출할 수도 있다고 생각해요 😃  
Reward의 정적 메서드인 `getReward`의 인자로 `일치하는 로또 숫자 개수`,`bonusBall 일치 여부`를 받아오게 함으로써, 변경할 수 있을 것 같아요 🤔

Reply:
음 확실히 불 필요할 수 있겠다는 생각이 드네요 굳이 상태를 저장 해 둘 필요가 없을 것 같아요!